### DISCUSSION TITLE
When is buffer free during interrupt driven transmission ? (SX1262)

### DISCUSSION THREAD

**Discussion by MrSniffer** [jgromes/RadioLib#1069] [Q&A]
If I move data into a buffer, and then do an interrupt driven transmission, e.g.

int16_t startTransmit(uint8_t* buffer, .... );

when the call to startTransmit returns: is the buffer free to be refilled, or must I wait for the transmit complete interrupt? I would like to refill the buffer while the SX1262 is sending out the previous packet. Or is the SX1262 sending it out from my userspace buffer?

**Comment by jgromes** [comment]
`startTransmit(uint8_t*` does not copy your buffer in the MCU, it sends out its contents to the radio. That's all being done in `startTransmit` which ends when the buffer has been sent and transmission begins. So you can modify the buffer during an ongoing transmission but only after `startTransmit` returns.

**Comment by HeadBoffin** [comment]
Consider if you need to fill the buffer that quickly, you are very likely breaching local legal laws regarding duty cycle - typically 1% maximum airtime. And if you are doing back to back transmissions on the same frequency, sooner or later it will look like a denial of service to other users ...
